py/desispec/workflow/proctable.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+
+import numpy as np
+import os
+
+from astropy.table import Table, vstack
+from collections import OrderedDict
+
+## Import some helper functions, you can see their definitions by uncomenting the bash shell command
+from desispec.workflow.exptable import default_exptypes_for_exptable
+
+from desispec.workflow.os import pathjoin
+
+def get_processing_table_column_defs(return_default_values=False, overlap_only=False, unique_only=False):
+    ## Define the column names for the internal production table and their respective datatypes, split in two
+    ##     only for readability's sake
+
+    colnames1 = ['EXPID'              , 'NIGHT' , 'OBSTYPE', 'CAMWORD'    , 'TILEID', 'CALIBRATOR']
+    coltypes1 = [np.ndarray           , int     , 'S10'    , 'S40'        , int     , np.int8     ]
+    coldefault1 = [np.ndarray(shape=0), 20000101, 'unknown', 'a0123456789', -99     , 0           ]
+
+    colnames2 = ['INTID', 'OBSDESC', 'SCRIPTNAME', 'JOBDESC', 'LATEST_QID', 'SUBMIT_DATE', 'STATUS']
+    coltypes2 = [int    , 'S16'    , 'S40'       , 'S12'    , int         ,  int        , 'S10'   ]
+    coldefault2 = [-99  , 'unknown', 'unknown'   , 'unknown', -99         , -99      , 'U'     ]
+
+    colnames3 =   ['INT_DEP_IDS'                  , 'LATEST_DEP_QID'               , 'ALL_QIDS'                     ]
+    coltypes3 =   [np.ndarray                     , np.ndarray                     , np.ndarray                     ]
+    coldefault3 = [np.ndarray(shape=0).astype(int), np.ndarray(shape=0).astype(int), np.ndarray(shape=0).astype(int)]
+
+    #     colnames1 = ['INTERNAL_ID', 'EXPID'    , 'NIGHT', 'INT_DEP_IDS', 'JOBNAME', 'TYPE']
+    #     coltypes1 = [ int         ,  list      ,  int    , list        , 'S30'    , 'S10']
+
+    #     colnames2 = ['LATEST_QID', 'LAST_SUBMIT_DATE', 'STATUS', 'LATEST_DEP_QID', 'ALL_QIDS']
+    #     coltypes2 = [ int        , 'S20'             , 'S10'   ,  list           ,  list      ]
+
+    colnames = colnames1 + colnames2 + colnames3
+    coldtypes = coltypes1 + coltypes2 + coltypes3
+    coldefaults = coldefault1 + coldefault2 + coldefault3
```

Comment:
I think it is a topic for a future update given that this is nominally working, but I'm concerned about the maintainability of these variable length arrays within a row of a column for tracking variable numbers of dependencies.  I think it will be more maintainable in the long run to keep one table for jobids, and a separate table or json structure for tracking the dependencies between the jobs.  Keeping the dependencies as a separate table would also make this more amenable to loading into an SQL database if needed for further scaling.  Future work.

Reply:
The concern is certainly warranted, but the scale of these things, especially for a daily production alone, is very small and a database seems like overkill.

As you say, it is something to investigate in the future. Especially if robustness starts to become questioned, then we will certainly want to look at shifting to a database.